### PR TITLE
configurable master server address

### PR DESCRIPTION
**Pull request by Duskhorn**
This pull request modifies the `server::defaultmaster()` function to take a string parameter to change the hardcoded value.
Moreover it adds a cubescript function `setmastername` to change the `mastername` variable

### CODE REVIEW

File: game/server.cpp [hunk 1]
```diff
@@ -158,6 +158,9 @@ VARN(updatemaster, allowupdatemaster, 0, 1, 1);
 SVAR(mastername, server::defaultmaster());
 VAR(masterport, 1, Port_Master, 0xFFFF);
 
+//this is ugly af, from the macro to the const_cast<char *>
+ICOMMAND(setmastername, "s", (const char *newname), mastername = const_cast<char *>(server::defaultmaster(newname)));
```
Comment (TheEgghead27) [inline game/server.cpp:162]: hey question what the HECK does this mean

this is horrifying but then again I don't browse code here ofter

**Review by TheEgghead27 — APPROVED** [review]

File: game/server.cpp (hunk 1 above)
Comment (Duskhorn) [reply game/server.cpp:162]: ICOMMAND creates a new cubescript "command", using an inline lambda

first parameter sets the command name
second parameter is the string with the parameter types. I think
third one is the parameter declarations
fourth one is the function body

**Comment by Duskhorn** [comment]
Now it should be ready to be merged. I kept the optional value in the function for now

**Comment by TheEgghead27** [comment]
hmm, should this PR be squashed to just the variable changes, so as to not add clutter to an already cluttered codebase?

**Comment by Duskhorn** [comment]
so you are suggesting to remove the function changes too?

**Comment by TheEgghead27** [comment]
i'd go a step further and reset the branch `git reset --hard master`, then make a new commit with just the necessary changes (attempting to cherry-pick would cause merge conflicts)

after that, `git push -f` to force the remote to accept the alternate history

**Comment by Duskhorn** [comment]
Ok fair, do I close the pull request and make a new one?

**Comment by TheEgghead27** [comment]
if you force push to the same github branch, you can keep this PR, but if you wanted to, you can start fully anew

**Comment by Duskhorn** [comment]
this should be cleaner and just easier. But I hate cubescript

**Pull request closed without merging**